### ISSUE TITLE
`padding_mode` in convolutions

### ISSUE THREAD

**Issue by SeKim12** [keras-team/keras#19988]
Hi! I wasn't able to find a quick way to specify different padding modes (e.g. `reflect` or `replicate`) for the Conv layers in Keras. For reference, [PyTorch](https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html) supports a `padding_mode` argument which gives users a bit more flexibility on the padding values, and it might be nice for Keras to support this kind of API as well. 

If such a feature doesn't already exist, I'd be happy to work on it. Thanks!

**Comment by mehtamansi29** [comment]
Hi @SeKim12 -

Currently Keras supports only [zeropadding](https://github.com/keras-team/keras/blob/v3.3.3/keras/src/layers/reshaping/zero_padding2d.py#L9) in the [convolution layer](https://github.com/keras-team/keras/blob/v3.3.3/keras/src/layers/convolutional/conv2d.py#L5). Keras doesn't have support for different padding modes(i.e reflect or replicate).

**Comment by SeKim12** [comment]
Right, I was wondering whether that might be a feature that could be included!

**Comment by mattdangerw** [comment]
@fchollet any thoughts on this? Adding more padding modes?

Looks like torch has 'reflect', 'replicate' or 'circular' and jax has 'circular'.

**Comment by mattdangerw** [comment]
Chatted with @fchollet. We would like to keep the Keras Conv offering simple here.

Very flexible padding could be done to the input in a separate layer before the conv layers. Can use https://keras.io/api/ops/numpy/#pad-function 

Closing this. But thanks very much for asking and filing!
